astroid/nodes/node_ng.py

```diff
@@ -38,6 +49,13 @@
 else:
     from astroid.decorators import cachedproperty as cached_property
 
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
```

Comment:
Mypy doesn't support `Self` yet.

Reply:
Do you want me to remove it? It does seem like the best fir.